### DISCUSSION TITLE
deck.gl SimpleMeshLayer: 3D line segments between meshes

### DISCUSSION THREAD

**Discussion by saravanabalagi** [visgl/deck.gl#5971] [Q&A]
I am trying to draw lines between meshes I rendered using SimpleMeshLayer. In particular, I have meshes that are translated above the ground at different heights. 

**Is there a way to add 3D line segments between such objects?**

Thoughts:

 - Adding a separate layer that can render 3D lines but it would require finding coords of the meshes and linking would be troublesome. 

 - Extending SimpleMeshLayer to allow take in an array of pairs of indices between which line segments need to be drawn. But I don’t know how to accomplish this. e.g. `getConnections (Function|Array, optional)` where Array is pairs of indices like [0,3]

What would make it more manageable? Has there been similar extensions to SimpleMeshLayer? Any pointers on how to do this would be helpful.

**Comment by ibgreen** [comment]
Connecting arbitrary meshes is a rather tricky problem and we don't have any plans around that. We are however working on "skirting" in the terrain loader, which makes each terrain tile have a vertical skirt which doesn't connect the meshes but prevents visible cracks. That might be similar to your use case, if I understood correctly. https://github.com/visgl/loaders.gl/pull/1662

**Reply by saravanabalagi** [reply]
Thanks I'll take a look.
